broker/src/main/java/io/camunda/zeebe/broker/system/partitions/ZeebePartitionHealth.java

```diff
@@ -46,6 +54,25 @@ private void updateHealthStatus() {
       healthReport = HealthReport.unhealthy(this).withMessage("Not enough disk space available");
     } else if (!servicesInstalled) {
       healthReport = HealthReport.unhealthy(this).withMessage("Services not installed");
+    } else if (!Objects.isNull(partitionContext.getTimeOfLastTransitionStep())
+        && System.currentTimeMillis()
+            > partitionContext.getTimeOfLastTransitionStep() + timeoutInMilliseconds
+        && partitionContext.getIsPartitionInTransition()) {
+      // we check if the time since last transition step is larger than timeout
+      // while if we're still in transition
+      LOG.error(
+          "Transition from {} on term {}, on transition step {} seems to be blocked."
+              + " Timeout configured is {}s.",
+          partitionContext.getCurrentRole(),
+          partitionContext.getCurrentTerm(),
+          partitionContext.getCurrentStepTransition(),
+          timeoutInMilliseconds / 1000); // display timeout in seconds
+
+      healthReport =
+          HealthReport.unhealthy(this)
+              .withMessage(
+                  "Partition is blocked in Transition Step %s"
+                      .formatted(partitionContext.getCurrentStepTransition()));
```

Comment:
If we put this logic that warns the user that the transition is stuck in the PartitionTransition don't we guarantee that this never gets shown to the user since that the transition step will be blocking the other methods in the class?

Reply:
If a partition step is stuck, only that step will no longer make progress and potentially block an entire actor and actor thread.
That doesn't mean that we can't ask `PartitionTransition` for a health report though! It's not the `PartitionTransition` that is "blocked".